File: Versions/v0-developing/src/OmniSphere.LiveStreamService/OmniSphere.LiveStreamService.Infrastructure/Exceptions/EntityNotFoundException.cs
namespace OmniSphere.LiveStreamService.Infrastructure.Exceptions;

public class EntityNotFoundException : Exception
{
    public EntityNotFoundException(string message) : base(message)
    {
    }

    public EntityNotFoundException(string name, string identifier) : base(GetEntityExceptionMessage(name, identifier))
    {
    }

    public EntityNotFoundException(string name, string identifier, string message, Exception ex) : base(
        GetEntityExceptionMessage(name, identifier) + $" {message}", ex)
    {
    }

    public EntityNotFoundException(string name, string identifier, Exception ex) : base(
        GetEntityExceptionMessage(name, identifier), ex)
    {
    }

    private static string GetEntityExceptionMessage(string identifier)
    {
        return $"{identifier} not found!";
    }

    private static string GetEntityExceptionMessage(string name, string identifier)
    {
        return $"Entity: {name}, on identifier: {identifier}, canno't be found!";
    }
}

File: Versions/v0.1-rebuild/src/User/src/main/java/omnisphere/microsservices/User/core/exceptions/EntityNotFoundException.java
package omnisphere.microsservices.User.core.exceptions;

public class EntityNotFoundException extends RuntimeException {
    public EntityNotFoundException() {
        super("This entity is not found!");
    }
    public EntityNotFoundException(String message) {
        super(message);
    }
    public EntityNotFoundException(String message, Throwable ex) {
        super(message, ex);
    }
}
